manageiq-operator/pkg/controller/manageiq/manageiq_controller.go

```diff
@@ -0,0 +1,317 @@
+	if err != nil { return err }
+
+	err = Createk8sResIfNotExist(cr, PostgresqlDeployment, &appsv1.Deployment{}, r)
+	if err != nil { return err }
+
+	return err
+}
+
+func GenerateOrchestratorResources(cr *miqv1alpha1.Manageiq, r *ReconcileManageiq) error {
+	OrchestratorDeployment := miqtool.NewOrchestratorDeployment(cr)
+	err := Createk8sResIfNotExist(cr, OrchestratorDeployment, &appsv1.Deployment{}, r)
+	if err != nil { return err }
+
+	return err
+}
+
+func GenerateSecrets(cr *miqv1alpha1.Manageiq, r *ReconcileManageiq) error {
+
+	PostgresqlSecret := miqtool.NewPostgresqlSecret(cr)
+	AppSecret := miqtool.AppSecret(cr)
+	TLSSecret := miqtool.TLSSecret(cr)
+
+	err := Createk8sResIfNotExist(cr, AppSecret, &corev1.Secret{}, r)
+	if err != nil { return err }
+	
+	err = Createk8sResIfNotExist(cr, TLSSecret, &corev1.Secret{}, r)
+	if err != nil { return err }
+	
+	err = Createk8sResIfNotExist(cr, PostgresqlSecret, &corev1.Secret{}, r)
+	if err != nil { return err }
+
+	return nil
+}
+
+func GenerateRbacResources(cr *miqv1alpha1.Manageiq, r *ReconcileManageiq) error {
+
+	HttpdServiceAccount := miqtool.HttpdServiceAccount(cr)
+	OrchestratorServiceAccount := miqtool.OrchestratorServiceAccount(cr)
+	AnyuidServiceAccount := miqtool.AnyuidServiceAccount(cr)
+	OrchestratorViewRoleBinding := miqtool.OrchestratorViewRoleBinding(cr)
+	OrchestratorEditRoleBinding := miqtool.OrchestratorEditRoleBinding(cr)
+
+	err := Createk8sResIfNotExist(cr, HttpdServiceAccount, &corev1.ServiceAccount{}, r)
+	if err != nil { return err }
+	err = Createk8sResIfNotExist(cr, OrchestratorServiceAccount, &corev1.ServiceAccount{}, r)
+	if err != nil { return err }
+	err = Createk8sResIfNotExist(cr, AnyuidServiceAccount, &corev1.ServiceAccount{}, r)
+	if err != nil { return err }
+	err = Createk8sResIfNotExist(cr, OrchestratorViewRoleBinding, &rbacv1.RoleBinding{}, r)
+	if err != nil { return err }
+	err = Createk8sResIfNotExist(cr, OrchestratorEditRoleBinding, &rbacv1.RoleBinding{}, r)
+	if err != nil { return err }
+
+	return nil 
+}
+
+func Createk8sResIfNotExist(cr *miqv1alpha1.Manageiq, res, restype metav1.Object, r *ReconcileManageiq) error {
+
+	reqLogger := log.WithValues("task: ", "create resource")
+	if err := controllerutil.SetControllerReference(cr, res, r.scheme); err != nil { return err }
+	resClient := res.(runtime.Object)
+	resTypeClient := restype.(runtime.Object)
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: res.GetName(), Namespace: res.GetNamespace()}, resTypeClient.(runtime.Object) )
+	if err != nil && errors.IsNotFound(err) {
+		reqLogger.Info("Creating ", "Resource.Namespace", res.GetNamespace(), "Resource.Name", res.GetName())
+		if err = r.client.Create(context.TODO(), resClient); err != nil { return err }
+		return nil
+	} else if err != nil { return err }
+	return nil
+}	
+
+func CleanUpOrchestratedDeployments( cr *miqv1alpha1.Manageiq, r *ReconcileManageiq) error {
```

Comment:
This function name seems strange. Maybe `RemoveManageiqResources` to contrast with the `Generate...Resources` functions?